src/orca-loader/Elements.res

```diff
@@ -623,17 +628,21 @@ let make = (
                   ~isForceSync=true,
                 )
                 ->then(json => {
-                  let dict = json->JSON.Decode.object->Option.getOr(Dict.make())
-                  let status = dict->getString("status", "")
-                  let returnUrl = dict->getString("return_url", "")
-                  Window.Location.replace(
-                    `${returnUrl}?payment_intent_client_secret=${clientSecret}&status=${status}`,
-                  )
-                  resolve()
+                  if redirect.contents === "always" {
+                    let dict = json->JSON.Decode.object->Option.getOr(Dict.make())
+                    let status = dict->getString("status", "")
+                    let returnUrl = dict->getString("return_url", "")
+                    Window.Location.replace(
+                      `${returnUrl}?payment_intent_client_secret=${clientSecret}&status=${status}`,
+                    )
+                    resolve(JSON.Encode.null)
+                  } else {
+                    resolve(json)
+                  }
                 })
-                ->catch(_ => {
+                ->catch(err => {
                   Window.Location.replace(url)
-                  resolve()
+                  resolve(err->Identity.anyTypeToJson)
```

Comment:
we need to add the same check here too ..

Reply:
updated